### PR TITLE
Normal vector node

### PR DESCRIPTION
**Pull request by Durman**
## Addressed problem description

I'm searching a way for setting direction by `direction` widget.
vector normal

It is not obvious how this should be implemented. By default the space has next view.

I gives opportunity to change it. Instead Z axis custom axis can be set in this case Z axis will replace Y axis.
Any suggestions are welcome.

## Preflight checklist

- [ ] Code changes complete.
- [ ] Code documentation complete.
- [ ] Documentation for users complete.
- [ ] Manual testing done. 
- [ ] Ready for merge.

### CODE REVIEW

File: nodes/vector/vector_normal_in.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+Modes = namedtuple('Modes', ['x', 'y', 'z', 'custom'])
+MODE = Modes('X', 'Y', 'Z', 'Custom')
+MODE_DIR = Modes(Vector((1, 0, 0)), Vector((0, 1, 0)), Vector((0, 0, 1)), None)
+
+QUAT_X90 = Quaternion((1, 0, 0), pi / 2)
+QUAT_Y270 = Quaternion((0, 1, 0), -pi / 2)
+
+
+def get_costume_norm(normal, space_normal, mode=MODE.z):
```
Comment (zeffii) [inline nodes/vector/vector_normal_in.py]: "custom"  not "costume"

**Comment by zeffii** [comment]
the 3dview rotation widget would be really cool for this, is that what you mean?

**Comment by Durman** [comment]
Yes 3d view widget would be even more flexible but direction vector is also cool.)

**Comment by zeffii** [comment]
i also don't know @Durman  : [

**Comment by Durman** [comment]
Not sure about this node yet.

**Pull request closed without merging**